### ISSUE TITLE
Reading BLOB data

### ISSUE THREAD

**Issue by snowtorious** [mariano/node-db-mysql#4]
I'm trying to read data stored in a BLOB column. The data is stored in a buffer and converting it to a string doesn't seem to be working:
<code>console.log('Data: ' + rows[0].data.toString('binary'));</code>

**Comment by mariano** [comment]
I don't see any issues with this. Done the following in a database:

```
CREATE TABLE articles(
    id INT NOT NULL AUTO_INCREMENT,
    body TEXT NOT NULL,
    package BLOB default NULL,
    PRIMARY KEY(id)
);

INSERT INTO articles(body, package) VALUES('My body', 'My package');
```

Then created the following script:

```
var mysql = require('db-mysql');
new mysql.Database({
    hostname: 'localhost',
    user: 'root',
    password: 'password',
    database: 'node'
}).on('ready', function() {
    this.query().select('*').from('articles').execute(function(rows) {
        console.log(rows[0]);
        console.log(rows[0].package.toString('binary'));
    });
}).connect();
```

And got the following output as a result:

```
$ node test.js                                                                                                                                                          
{ id: 1,                                                                                                                                                                                                  
  body: 'My body',                                                                                                                                                                                        
  package: <Buffer 4d 79 20 70 61 63 6b 61 67 65> }                                                                                                                                                       
My package                                                                                                                                                                                                
```

So when you say 'it doesn't seem to be working', what do you mean? What's the output for you?

**Comment by mariano** [comment]
Forgot to close this. Let me know how it goes @snowtorious

**Comment by snowtorious** [comment]
I've taken a closer look at my data. It seems that if a 00 value is read, the rest of the buffer gets filled with 00. For example, the first few bytes of the data in my database look like this:

<pre>
1F C2 8B 08 00 00 00 00 00 00 03 C2 B5 58 6D ...
</pre>


But the buffer on the node.js side of things looks like this:

<pre>
Buffer 1f c2 8b 08 00 00 00 00 00 00 00 00 00 00 00 ...
</pre>


Reading data without any 00 values does work.

**Comment by mariano** [comment]
Strange, since if that's the case it'd be an issue with v8's Buffer class.

Is the length of the 00-filled Buffer the same length you would've expected if the read was successfull? I'm trying to see if it gets filled with 00, or if it even truncates data.

**Comment by snowtorious** [comment]
Yes, it looks like the buffer is the same length as my data in the database.

**Comment by mariano** [comment]
Is there any way you could dump out the contents so i can insert them in my own MySQL?

**Comment by mariano** [comment]
@snowtorious: Just fixed this issue in 0.4.8, so reinstall db-mysql. Was pretty easy to replicate by inserting CONCAT('Hello', CHAR(0), 'World') to a BLOB
